### PR TITLE
Add rpm-ostree support 

### PR DESCRIPTION
**Pull request by adisbladis**
Currently not working:
- [ ] I broke non-ostree setups with the whole `chattr` dance
- [ ] Fix auto-startup (currently broken)

Current steps to run nix on an rpm-ostree distro:
- `rpm-ostree install foo.rpm`
- `systemctl reboot`
- `systemctl start nix-daemon` <- Fails because it's lacking correct SELinux labels
- `restorecon -FR /nix`
- `systemctl start nix-daemon`

### CODE REVIEW

File: rootfs/usr/lib/systemd/system/mkdir-rootfs@.service [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Enable mount points in / for ostree
+DefaultDependencies=no
+ConditionPathExists=!%f
+
+[Service]
```
Comment (adisbladis) [inline rootfs/usr/lib/systemd/system/mkdir-rootfs@.service:6]: Ideally this entire unit would not even exist on non ostree distros.

File: rootfs/usr/share/nix/nix-setup [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if test -e /var/nix/store; then
+    exit 0
+fi
+
+# Create the stoure
```
Comment (Yajo) [inline rootfs/usr/share/nix/nix-setup:8]: Typo?

```suggestion
# Create the store
```

File: rootfs/usr/share/nix/nix-setup [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+fi
+if ! [[ "@channelURL@" = "" || "@channelName@" = "" ]] && ! test -e /root/.nix-channels; then
+    echo "@channelURL@ @channelName@" > /root/.nix-channels
+fi
+
+if test -e /sys/fs/selinux; then
+    semodule -i "/usr/share/selinux/packages/nix.pp"
+
+    # # Relabel the SELinux security context
+    # restorecon -FR /nix
```
[the remark this answers is not in the document]
Comment (Yajo) [reply rootfs/usr/share/nix/nix-setup:26]: I have two ideas.

1. You can do this *after* you create the `/nix` dir, but *before* you do the bind mount. In that moment, `/nix` should be empty, so running this command should be a matter of milliseconds.
2. You can do the recursive restorecon just once in `/var/nix` directly. Then, after each mount, do a *not recursive* restorecon for `/nix` (the dir contents should not change, so no need to recurse each time, right?)

Does any of them make sense?

File: rootfs/usr/lib/systemd/system/mkdir-rootfs@.service (hunk 1 above)
Comment (Yajo) [reply rootfs/usr/lib/systemd/system/mkdir-rootfs@.service:6]: Maybe you could link it to the existence of `ostree-remount.service`?

**Comment by adisbladis** [comment]
I'm going to close this as I don't like the jank it introduces.

**Pull request closed without merging**